src/connectors/impls/clickhouse.rs

```diff
@@ -0,0 +1,491 @@
+    const HOST: &'static str = "localhost";
+    const PORT: u16 = 9000;
+}
+
+impl ConfigImpl for ClickhouseConfig {}
+
+#[derive(Clone, Copy, Deserialize)]
+#[serde(rename_all = "snake_case")]
+enum Compression {
+    None,
+    Lz4,
+}
+
+impl Default for Compression {
+    fn default() -> Compression {
+        Compression::None
+    }
+}
+
+impl Display for Compression {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            Compression::None => "none",
+            Compression::Lz4 => "lz4",
+        }
+        .fmt(f)
+    }
+}
+
+#[derive(Deserialize)]
+struct Column {
+    name: String,
+    #[serde(rename = "type")]
+    type_: DummySqlType,
+}
+
+pub(crate) struct ClickhouseSink {
+    db_url: String,
+    handle: Option<ClientHandle>,
+    table: String,
+    columns: Vec<(String, DummySqlType)>,
+}
+
+#[async_trait::async_trait]
+impl Sink for ClickhouseSink {
+    async fn connect(&mut self, ctx: &SinkContext, _attempt: &Attempt) -> Result<bool> {
+        let pool = Pool::new(self.db_url.as_str());
+        let handle = match pool.get_handle().await {
+            Ok(handle) => handle,
+            Err(e) => {
+                return match e {
+                    CError::Driver(_) | CError::Io(_) | CError::Connection(_) => {
+                        ctx.notifier.connection_lost().await?;
+                        Ok(false)
+                    }
+                    _ => Err(Error::from(e)),
+                }
+            }
+        };
+
+        self.handle = Some(handle);
+
+        Ok(true)
+    }
+
+    async fn on_event(
+        &mut self,
+        _input: &str,
+        event: Event,
+        _ctx: &SinkContext,
+        _serializer: &mut EventSerializer,
+        _start: u64,
+    ) -> Result<SinkReply> {
+        let handle = self
+            .handle
+            .as_mut()
+            .ok_or_else(|| Error::from(ErrorKind::NoClickHouseClientAvailable))?;
+
+        let mut block = Block::with_capacity(event.len());
+
+        for value in event.value_iter() {
+            let row = Self::clickhouse_row_of(&self.columns, value)?;
+            block.push(row)?;
+        }
+
+        debug!("Inserting block:{:#?}", block);
+        handle.insert(&self.table, block).await?;
+
+        Ok(SinkReply::NONE)
+    }
+
+    fn auto_ack(&self) -> bool {
+        true
+    }
+}
+
+impl ClickhouseSink {
+    fn clickhouse_row_of(
+        columns: &[(String, DummySqlType)],
+        input: &tremor_value::Value,
+    ) -> Result<Vec<(String, clickhouse_rs::types::Value)>> {
+        let mut rslt = Vec::new();
+
+        let object = input
+            .as_object()
+            .ok_or_else(|| Error::from(ErrorKind::ExpectedObjectEvent(input.value_type())))?;
+
+        for (column_name, expected_type) in columns.iter() {
+            // If the value is not present, then we can replace it by null.
+            const NULL: Value = Value::Static(StaticNode::Null);
```

Comment:
you can use `Value::const_null()` instead :) it's a const fn doing exactly this

Reply:
Nice catch. It is fixed by commit [6ff650f](https://github.com/tremor-rs/tremor-runtime/pull/1538/commits/6ff650f9c429b085a2e708f12c9b6c76753c3ed5).